### DISCUSSION TITLE
Daggerverse: should be able to set `name` at `dagger develop`

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#12395] [Ideas]
Right now you can only set the module's name at `dagger init` either with `--name` or by default the name of the containing directory will be used.

I found this odd and expected to be able to do this:

```
dagger init
dagger develop --sdk=go --name=foobar --source=.
```
> Side note: since `dagger init` assumes `.` I'd want `dagger develop` to assume `source=.` also. 
https://github.com/dagger/dagger/issues/6963

Instead, I got this error with the above:
```
dogfish ➤ dagger init
09:47:50 WRN no LICENSE file found; generating one for you, feel free to change or remove license=Apache-2.0
Initialized module dogfish in .
dogfish ➤ dagger develop --sdk go --name foobar --source .
Error: unknown flag: --name
```
I don't know if we need a name with `dagger init` for technical reasons, but I'd leave it out if we don't. For example, I can `git init` without setting a `remote`.

**Comment by helderco** [comment]
Related to https://github.com/dagger/dagger/issues/6119

**Comment by helderco** [comment]
That's a good point. Adding `--name` to `develop`, I think the rule is to **not set** `--name` by default unless you also set `--sdk`.

**Comment by helderco** [comment]
Can even change the name if you have an existing module with a name but no sdk.
